pkg/agent/nodeportlocal/portcache/port_table_windows_test.go

```diff
@@ -0,0 +1,151 @@
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	allNPLPorts := []rules.PodNodePort{
+		{
+			NodePort: nodePort1,
+			PodPort:  1001,
+			PodIP:    podIP,
+			Protocol: "tcp",
+		},
+		{
+			NodePort: nodePort1,
+			PodPort:  1001,
+			PodIP:    podIP,
+			Protocol: "udp",
+		},
+		{
+			NodePort: nodePort2,
+			PodPort:  1002,
+			PodIP:    podIP,
+			Protocol: "udp",
+		},
+	}
+
+	mockPortRules.EXPECT().AddRule(nodePort1, podIP, 1001, "tcp")
+	mockPortRules.EXPECT().AddRule(nodePort1, podIP, 1001, "udp")
+	mockPortRules.EXPECT().AddRule(nodePort2, podIP, 1002, "udp")
+
+	syncedCh := make(chan struct{})
+	err := portTable.RestoreRules(allNPLPorts, syncedCh)
+	require.NoError(t, err)
+}
+
+func TestDeleteRule(t *testing.T) {
+	var err error
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockPortRules := rulestesting.NewMockPodPortRules(mockCtrl)
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	npData := &NodePortData{
+		NodePort: startPort,
+		PodIP:    podIP,
+		PodPort:  1001,
+		Protocol: ProtocolSocketData{
+			Protocol: "tcp",
+		},
+	}
+
+	tcs := []struct {
+		name         string
+		testFunction bool
+	}{
+		{
+			name:         "Delete Rule",
+			testFunction: true,
+		},
+		{
+			name:         "Delete Rules For Pod",
+			testFunction: false,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			portTable.addPortTableCache(npData)
+			mockPortRules.EXPECT().DeleteRule(startPort, podIP, 1001, "tcp")
+			if tc.testFunction {
+				err = portTable.DeleteRule(podIP, 1001, "tcp")
+			} else {
+				err = portTable.DeleteRulesForPod(podIP)
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestAddRule(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	mockPortRules := rulestesting.NewMockPodPortRules(mockCtrl)
+	mockPortOpener := portcachetesting.NewMockLocalPortOpener(mockCtrl)
+	portTable := newPortTable(mockPortRules, mockPortOpener)
+	podPort := 1001
+	tcs := []struct {
+		name        string
+		nodePort    int
+		protocol    string
+		expectedErr string
+	}{
+		{
+			name:     "valid entry for rule",
+			nodePort: startPort,
+			protocol: "udp",
+		},
+		{
+			name:        "duplicate rule",
+			nodePort:    startPort + 1,
+			protocol:    "udp",
+			expectedErr: "existing Windows Nodeport entry for",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.expectedErr != "" {
+				_, err := portTable.AddRule(podIP, podPort, tc.protocol)
+				assert.ErrorContains(t, err, tc.expectedErr)
+			} else {
+				mockPortRules.EXPECT().AddRule(tc.nodePort, podIP, podPort, tc.protocol)
+				nodePort, err := portTable.AddRule(podIP, podPort, tc.protocol)
+				require.NoError(t, err)
+				assert.Equal(t, tc.nodePort, nodePort)
+			}
+		})
```

Comment:
Better to refactor it as below:
```suggestion
			if tc.expectedErr == "" {
			    mockPortRules.EXPECT().AddRule(tc.nodePort, podIP, podPort, tc.protocol)
			}
			nodePort, err := portTable.AddRule(podIP, podPort, tc.protocol)
			if tc.expectedErr != "" {
				assert.ErrorContains(t, err, tc.expectedErr)
			} else {
				require.NoError(t, err)
				assert.Equal(t, tc.nodePort, nodePort)
			}
		})
```